server/src/controllers/auth.controllers.ts

```diff
@@ -0,0 +1,262 @@
+import { Request, RequestHandler, Response } from 'express';
+import { User } from '../models/user.model';
+import { ResetPwdToken } from '../models/auth.model';
+import { UserSchema, UserDocument, UserAuthSchema } from '../types/user.types';
+import {
+  hashInput,
+  validateInput,
+  generateRandomToken,
+  generateJwt,
+  verifyGoogleToken,
+  verifyJwt,
+} from '../utils/auth.utils';
+import {
+  CredentialsSchema,
+  GooglePayloadSchema,
+  ResetPwdTokenSchema,
+  ResetPwdTokenDocument,
+} from '../types/auth.types';
+import { OAuth2Client } from 'google-auth-library';
+import { sendTemplateEmail } from '../utils/email.utils';
+import { PWD_RESET_TOKEN_TTL } from '../constants';
+
+const sendResponse = (
+  res: Response,
+  statusCode: number,
+  payload: UserDocument | GooglePayloadSchema | UserAuthSchema | string
+) => {
+  return res.status(statusCode).json(payload);
+};
+
+const login: RequestHandler = async (req: Request, res: Response): Promise<void> => {
+  const credentials: CredentialsSchema = req.body;
+
+  if (
+    Object.keys(credentials).length > 0 &&
+    credentials.email &&
+    credentials.password &&
+    credentials.isRememberPassword
+  ) {
+    try {
+      const userRecord: UserDocument | null = await User.findOne({
+        email: credentials.email.toLowerCase(),
+      });
+
+      if (userRecord && (await validateInput(credentials.password, userRecord.password))) {
```

Comment:
hmmm i think this `await validateInput` can be in a variable? so it can be easier to read